app/src/lib/stores/app-store.ts

```diff
@@ -3279,7 +3279,13 @@ export class AppStore extends TypedBaseStore<IAppState> {
       const defaultRemote = await getDefaultRemote(repository)
       // TODO: I think we could skip this fetch if we know that we have the branch locally
       // already. That way we'd match the behavior of checking out a branch.
-      if (defaultRemote) {
+      const gitStore = this.getGitStore(repository)
+      const localBranchName = `pr/${pullRequest.number}`
```

Comment:
> (could i get a second pair of 👀 on this?)

@nerdneha @agisilaos I don't believe this isn't the right name to lookup for the local branch **when the PR is associated with a branch in the repository, not a fork**.

Let me try and explain my thinking:

 - before we hit this code path, we know the PR comes from the original repository and not a fork
 - after this new code, we continue to git checkout `head.ref`, and not a new `pr/{id}` ref like we would for a fork PR
 - so `doesBranchExists` should be looking for `head.ref` as the local branch

I think that might affect your suggestion to eliminate the duplication, or change it at least.

Reply:
Actually, there's definitely more nuance to this. Let me poke at some `desktop/desktop` PRs and clarify what checks we should be doing here.

Reply:
Here's the rough approach that I think we need to do here (avoiding a patch to ensure we're all on the same page about the steps to follow):

```ts
if (isRefInThisRepo) {
  const defaultRemote = await getDefaultRemote(repository)
  // if we don't have a default remote here, it's probably going to just crash and burn
  // on checkout, but that's okay
  if (defaultRemote) {
    // the remote ref will be something like `origin/my-cool-branch` so we should look
    // for that rather than a local branch named `my-cool-branch`
    const remoteRef = `${defaultRemote.name}/${head.ref}`
    const gitStore = this.getGitStore(repository)
    // branch.name here is the fully-qualified ref - perhaps that isn't quite clear from the name?
    const remoteRefExists =
      gitStore.allBranches.find(branch => branch.name === remoteRef) != null

    // only try a fetch here if we can't find the ref
    if (!remoteRefExists) {
      await this._fetchRemote(
        repository,
        defaultRemote,
        FetchType.UserInitiatedTask
      )
    }
  }

  // now we should rely on Git's behaviour for finding the remote branch if
  // no local branch matching the name exists, so no change necessary here
  await this._checkoutBranch(repository, head.ref)
} else if (head.gitHubRepository != null) {
```